### DISCUSSION TITLE
[Q] Bundle size is not decreasing in angular prod build

### DISCUSSION THREAD

**Discussion by GillesVercammen** [exceljs/exceljs#2393] [Q&A]
Hi, 

we are using exceljs in our project but we noticed that the package size is huge. 
In dev it is not a real problem but for the production builds, the size is not decreasing. 
We are like the ease of use of the package, so we don't actually want to change to another implementation. 
But if we aren't able to get it smaller, it will be necessary.

Have any of you guys have an idea why the package size isn't decreasing for our prod build, or if we need to do something in order to achieve i?

These are the commands used to analyze the bundle
```
npm run build -- --stats-json && npm run analyze-bundle
npm run build:prod -- --stats-json && npm run analyze-bundle

```

Below are 2 images comparing the node modules size and content of the dev and prod bundle, as you can see the size of exceljs stays 1.08mb in both bundles.

dev:
![devbundle](https://user-images.githubusercontent.com/22342373/103407486-25051b00-4b5f-11eb-8f68-1526a6a32092.PNG)

prod:
![prodbundle](https://user-images.githubusercontent.com/22342373/103407348-a9a36980-4b5e-11eb-9734-a8c0e0f58c80.PNG)

**Comment by KingDarBoja** [comment]
Same issue here, guess the library is not optimized for tree-shaking?

**Comment by 4javier** [comment]
https://github.com/exceljs/exceljs/issues/1520

**Comment by sureshHARDIYA** [comment]
Same issue for me. Anyone found any solutions?

**Comment by nev-21** [comment]
Same issue here. Please help

**Comment by dmytro-afonin** [comment]
Same issue

**Comment by rogeriopgp** [comment]
Same issue here

**Comment by barbalex** [comment]
Same issue. Eceljs is BY FAR the single biggest part of my bundles. About 1 MB!
